modules/ServerAPI/src/server/python/acas_ldclient/acasldclient.py

```diff
@@ -0,0 +1,189 @@
+"""
+Interface for fetching data for ACAS API
+"""
+from ldclient.client import LDClient
+import argparse
+import json
+import os, sys
+
+def get_parser():
+    """
+    @return:
+        A command-line parser with switches and defaults defined.
+    @rtype:
+        argparse.ArgumentParser
+
+    """
+    # Get an argsparse parser, and populate it
+    script_usage = "python {} [-t] (optional: none curerntly) [Test(s) to run]".format(os.path.basename(sys.argv[0]))
+    script_desc = "Run lambda_function tests"
+    parser = argparse.ArgumentParser(
+        usage=script_usage,
+        description=script_desc
+    )
+    parser.add_argument(
+        '--ldserver',
+        '-l',
+        help='Live Design Server',
+        dest='ld_server',
+        required=True
+    )
+    parser.add_argument(
+        '--user',
+        '-u',
+        help='LD client user',
+        dest='username',
+        required=True
+    )
+    parser.add_argument(
+        '--password',
+        '-p',
+        help='LD client password',
+        dest='password',
+        required=True
+    )
+    parser.add_argument(
+        '--method',
+        '-m',
+        help='Method to call',
+        dest='method',
+        required=True
+    )
+    parser.add_argument(
+        '--args',
+        '-a',
+        nargs='*',
+        help='arguments to the method',
+        dest='args',
+        required=False
+    )
+    return parser
+
+def list_groups(client, test):
+    print test
+    return client.list_groups()
+
+def auth_check(client, username, password):
+    auth_return = client.auth_check(username = username, password = password)
+    return auth_return
+
+def ld_group_to_acas_role(group):
+    # u'LDAP_Auto_ROLE_SEURAT-USERS'
+    # u'Project_ProjectX_Administrator'
+    split = group["name"].split('_')
+    ls_type = split[0]
+    ls_kind = split[1]
+    role_name = "_".join(split[2:len(split)])
```

Comment:
We've discussed this elsewhere, so I'm OK with it as part of this merge, but I think we should get away from doing string manipulation like this to map roles if possible.

Reply:
Agreed